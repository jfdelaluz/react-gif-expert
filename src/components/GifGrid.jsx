import { useFetchGifs } from "../hooks/useFetchGifs";
import { GifItem } from "./GifItem";

export const GifGrid = ({ category, onRemoveCategory }) => {

  const { images, isLoading } = useFetchGifs( category );

  return (
    <>
      <h3>{ category }</h3>
      <div className="card-grid">
        { isLoading && <h2>Loading...</h2> }
        {
          images.map( ( image ) => (
              <GifItem
                key={ image.id }
                { ...image }
              />
            )
          )
        }
      </div>
    </>
  )
}
