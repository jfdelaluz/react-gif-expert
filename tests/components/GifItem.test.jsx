import { render, screen } from '@testing-library/react';
import { GifItem } from '../../src/components/GifItem';

describe('Tests - GifItem Component', () => {

  const title = 'Test Title';
  const url = 'http://www.testurl.com/test-image';

  test('Should match snapshot', () => {
    const { container } = render( <GifItem url={ url } title={ title } /> );
    expect( container ).toMatchSnapshot();
  });

  test('Should display image with correct URL and ALT', () => {
    render( <GifItem url={ url } title={ title } /> );

    const { src, alt } = screen.getByRole('img');
    expect( src ).toBe( url );
    expect( alt ).toBe( title );
  });

  test('should display title in component', () => {
    render( <GifItem url={ url } title={ title } /> );
    expect( screen.getByText( title ) ).toBeTruthy();
  });
});

