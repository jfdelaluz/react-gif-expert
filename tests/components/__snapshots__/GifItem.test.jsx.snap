// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests - GifItem Component Should match snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Test Title"
      src="http://www.testurl.com/test-image"
    />
    <p>
      Test Title
    </p>
  </div>
</div>
`;
